analyzer_test.go

```diff
@@ -53,9 +54,10 @@ func testAnalyzer(t *testing.T, when spec.G, it spec.S) {
 		h.AssertNil(t, err)
 
 		analyzer = &lifecycle.Analyzer{
-			Buildpacks: []lifecycle.GroupBuildpack{{ID: "metadata.buildpack"}, {ID: "no.cache.buildpack"}, {ID: "no.metadata.buildpack"}},
-			LayersDir:  layerDir,
-			Logger:     &log.Logger{Handler: &discard.Handler{}},
+			Buildpacks:  []lifecycle.GroupBuildpack{{ID: "metadata.buildpack"}, {ID: "no.cache.buildpack"}, {ID: "no.metadata.buildpack"}},
+			LayersDir:   layerDir,
+			Logger:      &log.Logger{Handler: &discard.Handler{}},
+			PlatformAPI: api.MustParse("0.5"),
```

Comment:
A pattern we've been trying to follow in tests of other phases (build?) is to have the "default" platform API version be the latest one, and have separate contexts for the earlier platform APIs. In theory, this would make it easier to drop support for earlier APIs if/when the time comes.